### ISSUE TITLE
"import tools" in my codebase imports detectron2's tools rather than my own library.

### ISSUE THREAD

**Issue by danielgordon10** [facebookresearch/detectron2#3934]
It's pretty common practice for pip libraries to put all their code inside a folder with the same name in order to namespace all packages to the outermost unique library name. In the detectron2 repo, there are a few folders that violate this, notably `tools` and `projects`. I would like to understand why they are set up this way rather than living inside the `detectron2` folder. Is there a way I can fix this by modifying the setup.py process? Or will I have to move or rename the folders in my fork and risk merge issues down the line?

## Instructions To Reproduce the Issue:
In my codebase

`from tools import foo`

## Expected behavior:

Imports my script named `foo` living in `<root>/tools/foo.py`

## Actual behavior

Python says `"foo" is an unknown symbol`

**Comment by ppwwyyxx** [comment]
Take https://github.com/numpy/numpy/tree/main/tools for example. It is not a wrong thing to have these directories.

I think the problem is either because 
* you didn't install detectron2 properly through pip (but with hacks like PYTHONPATH). 
   * in this case you should install it properly
* Or, you write new code inside detectron2's root directory
   * in this case you should develop your project in your own directory

**Comment by danielgordon10** [comment]
We are installing via creating a wheel and then pip installing that wheel.
I actually noticed the change may have aligned with detectron2 adding the `__init__.py` inside the tools directory. The numpy example you posted does not have an init file in their tools directory. We did not see this problem before the `__init__.py` file was created.

**Comment by ppwwyyxx** [comment]
Thanks for the info. I don't know there is such a file and it looks like a hack to me from the log. 
Having that file can explain the issue for `tools`, but I don't think `projects` will have similar issue.

For `tools`, I think adding it to the exclude here:
https://github.com/facebookresearch/detectron2/blob/710e7795d0eeadf9def0e7ef957eea13532e34cf/setup.py#L154
probably will fix it.

**Comment by xwang233** [comment]
Hi @ppwwyyxx , I'm pretty sure when I install `numpy`, they don't copy their `tools` folder to pollute the global python site-packages import path. See https://github.com/pytorch/benchmark/issues/970

**Comment by ppwwyyxx** [comment]
@xwang233 I agree with you, and I think my comment above at https://github.com/facebookresearch/detectron2/issues/3934#issuecomment-1028557008 will probably fix it

**Comment by tkkcc** [comment]
you need to clean rebuild it, s.t. remove build and egg folder first.
and use `pip install` without `-e`
@gallegi
